### PR TITLE
feat(parser): add support for Lambda Function URL

### PR DESCRIPTION
**Pull request by ran-isenberg**
**Issue number: #1425**

### Changes
added envelope and model for lambda function url as described at https://github.com/awslabs/aws-lambda-powertools-python/issues/1425

fixed a bug in api gwv2 model where cognitoIdentity wasnt optional.

I was unable to fix the docs for the parser due to a linter bug, will open an issue.

### User experience

Can now use an envelope for lambda function url or extend the model.

<details>
<summary>Is this a breaking change? No.</summary>

**RFC issue number**:
#1425
Checklist:

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by ran-isenberg** [comment]
@heitorlessa  i wasnt able to push the docs update.
They pass the docs-linter stage but fail on the pre-commit on this error:
docs/utilities/parser.md:586 MD043/required-headings/required-headers Required heading structure [Context: "## Getting started"]

**Comment by leandrodamascena** [comment]
Hi @ran-isenberg.. I'm refactoring all examples and for now you can add this comment after the description line..

`
`

like this: https://github.com/awslabs/aws-lambda-powertools-python/commit/7be425e05f2c193a20c1558de867dc39b81cabb3#diff-aa083264efcaf3125cc0560e9b88cce2c69e4becd8ea39211057634ec3f56bd9R6

**Comment by heitorlessa** [comment]
Thanks a lot @ran-isenberg !! As @leandrodamascena shared, this is the new doc linting to enforce a consistent style, you can ignore for now with that string at the top of the file since we're refactoring all of it (structure, snippets to be way more realistic, run as-is). 

@leandrodamascena could you please fix that mypy issue by installing `snappy` as a dev dependency and ignoring the lack of types for snappy? We missed in the last PR as I accidentally didn't allow GH Actions to run (part of the new repo hardening)

**Comment by ran-isenberg** [comment]
> Hi @ran-isenberg.. I'm refactoring all examples and for now you can add this comment after the description line..
> 
> `<!-- markdownlint-disable MD043 --> `
> 
> like this: [7be425e#diff-aa083264efcaf3125cc0560e9b88cce2c69e4becd8ea39211057634ec3f56bd9R6](https://github.com/awslabs/aws-lambda-powertools-python/commit/7be425e05f2c193a20c1558de867dc39b81cabb3#diff-aa083264efcaf3125cc0560e9b88cce2c69e4becd8ea39211057634ec3f56bd9R6)

Tried it but it doesnt work for me, cant commit. I added this linter to my own project and just removed this annoying check from the config :)

**Comment by ran-isenberg** [comment]
pushed the docs ;)

**Comment by ran-isenberg** [comment]
@heitorlessa  it failed on the snappy stuff. I updated from develop branch, hopefully it will work now

**Comment by heitorlessa** [comment]
Yes it was fixed on #1446, my other PR was impacted and working after the
merge

**Comment by heitorlessa** [comment]
@rubenfonseca assigned it to you as you fixed the quirks of Function URL.

@ran-isenberg CI passed. Ruben is back on Monday from holidays and should get this reviewed and merged next week.

Thank you again

**Comment by rubenfonseca** [comment]
Just rebased with master before the review

File: aws_lambda_powertools/utilities/parser/envelopes/lambda_function_url.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            Lambda event to be parsed
+        model : Type[Model]
+            Data model provided to parse after extracting data using envelope
+
+        Returns
+        -------
+        Any
+            Parsed detail payload with model provided
+        """
+        logger.debug(f"Parsing incoming data with Lambda function URL {LambdaFunctionUrlModel}")
```
Comment (rubenfonseca) [inline aws_lambda_powertools/utilities/parser/envelopes/lambda_function_url.py]:
```suggestion
        logger.debug(f"Parsing incoming data with Lambda function URL model {LambdaFunctionUrlModel}")
```

File: docs/utilities/parser.md [hunk 2]
```diff
@@ -166,6 +167,7 @@ Parser comes with the following built-in models:
 | **SnsModel**                    | Lambda Event Source payload for Amazon Simple Notification Service |
 | **APIGatewayProxyEventModel**   | Lambda Event Source payload for Amazon API Gateway                 |
 | **APIGatewayProxyEventV2Model** | Lambda Event Source payload for Amazon API Gateway v2 payload      |
+| **LambdaFunctionUrlModel**      | Lambda Event Source payload for Lambda function URL payload        |
```
Comment (rubenfonseca) [inline docs/utilities/parser.md]:
```suggestion
| **LambdaFunctionUrlModel**      | Lambda Event Source payload for Lambda Function URL payload        |
```

**Review by rubenfonseca — CHANGES_REQUESTED** [review]
Hi @ran-isenberg thank you for this! Just left two small suggestions. Let me know if it makes sense so we can merge.

**Review by rubenfonseca — APPROVED** [review]

**Comment by leandrodamascena** [comment]
> Hi @ran-isenberg thank you for this! Just left two small suggestions. Let me know if it makes sense so we can merge.

Hi!! Just in case of better understanding of the project I did a double check and everything seems fine to merge. I ran these changes locally and it went fine.

**Comment by ran-isenberg** [comment]
@heitorlessa  waiting for the merge :)

**Comment by heitorlessa** [comment]
@rubenfonseca will be merging ;) Thanks a lot everyone!!

**Comment by rubenfonseca** [comment]
Thanks again Ran! Can't wait to get this out soon :)

**Pull request merged**